src/txn/txn_recover.c

```diff
@@ -155,6 +156,12 @@ __txn_system_op_apply(WT_RECOVERY *r, WT_LSN *lsnp, const uint8_t **pp, const ui
     session = r->session;
     conn = S2C(session);
 
+    WT_RET(__wt_scr_alloc(session, 0, &lsn_str));
+    if ((ret = __wt_lsn_string(session, lsnp, lsn_str)) != 0) {
+        __wt_scr_free(session, &lsn_str);
```

Comment:
Do you want to output any kind of message? Or you could do something like:
```
bool have_lsn_str;

WT_RET(wt_scr_alloc...)
have_lsn_str = false;
WT_ERR(wt_lsn_string(...))
have_lsn_str = true;

....
err:
    if (have_lsn_str)
        print error message
    wt_scr_free...
```

Reply:
Makes sense. I've used the `have_lsn_str` approach in [68f7c9f](https://github.com/wiredtiger/wiredtiger/pull/11056/commits/68f7c9f8469c24972a0283e57b4fa5381f5433b4) and I'll merge once PR testing passes.
Thanks for all your feedback!

Reply:
edit: I got caught out by an uninitialized variable in the prior implementation. Rather than handle the extra complexity I'll print a simple error message in the fail cases. `__wt_lsn_string` failing should be rare enough that we won't see this fire.